### DISCUSSION TITLE
Add time window constraint for vehicles too

### DISCUSSION THREAD

**Discussion by lvalladares** [google/or-tools#3592] [Routing (and legacy CP) questions] [answered]
Hi!

I'm trying to solve a classic vehicle routing problem with a time constraint, I've several pickups that need to be done in the day, each one with a time window, and following this [example](https://developers.google.com/optimization/routing/vrptw) it works well.

But now I need to add time windows to vehicles too, so for example I have a vehicle that is only available from 8 AM to 2 PM and another is available from 10 AM to 5 PM. I've tried this: 

```python
# Add time window restriction
for i in range(len(self.job_requests)):
    job = self.job_requests[i]
    if job.time_window_start > 0:
        index = self.manager.NodeToIndex(i)
        self.time_dimension.CumulVar(index).SetRange(job.time_window_start, job.time_window_end)

for idx, vehicle in enumerate(self.vehicles):
    index = self.routing.Start(idx)
    self.time_dimension.CumulVar(index).SetRange(vehicle.route_time_start, vehicle.route_time_end)
```

for adding a time window first to jobs and then to vehicles but is not working, the vehicle time window is only being considered in the first node of the route.

**Comment by Mizux** [comment, marked as answer]
same with the `End` node ?
```py
for idx, vehicle in enumerate(self.vehicles):
    index = self.routing.End(idx)
    self.time_dimension.CumulVar(index).SetRange(vehicle.route_time_start, vehicle.route_time_end)
```

**Reply by lvalladares** [reply]
Yep, it worked. IDK how I missed that.
